smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSink.java

```diff
@@ -266,6 +267,11 @@ private Object getKey(Message<?> message,
             return ((Record) message.getPayload()).key();
         }
 
+        // Then, check if the message contains incoming metadata from which we can forward the key
+        if (configuration.getForwardKey() && message.getMetadata(IncomingKafkaRecordMetadata.class).isPresent()) {
+            return message.getMetadata(IncomingKafkaRecordMetadata.class).get().getKey();
```

Comment:
What do we need to do if it's null. Should be used the configured one or null?

Reply:
My initial implementation did a fallback on configured `key` if both `forward-key` and `key` were configured and incoming record key was null (or `IncomingKafkaRecordMetadata` wasn't there).
I now think that it makes more sense.